2.2: Linux, Build fails  with error: changes meaning of ‘route’ from ‘struct caspar::core::route’

On git revision ec623f08495e2315aabfe4022561da95ab6dc088
Debian 9.3 with all dependencies installed.
See attached buildlog

[buildlog.txt](https://github.com/CasparCG/server/files/1728080/buildlog.txt)

Dockerbuild:
Sending build context to Docker daemon  3.692MB
Step 1/15 : FROM casparcg/base:1.1.0 as build-casparcg
1.1.0: Pulling from casparcg/base
18c8f805be45: Pull complete 
f7869fb3b98c: Pull complete 
f75000414ffb: Pull complete 
c27c5092058d: Pull complete 
add65c7404ea: Pull complete 
077d47163a81: Pull complete 
e2bf7912836d: Pull complete 
7dff99104cbf: Pull complete 
9d49875aaecb: Pull complete 
Digest: sha256:c3eca1b6c2c266e9779fbc80b8f11da2f38f6edf2b8b280a275709a1655dbf56
Status: Downloaded newer image for casparcg/base:1.1.0
 ---> e342a6036584
Step 2/15 : COPY --from=casparcg/boost:1.66 /opt/boost /opt/boost
1.66: Pulling from casparcg/boost
18c8f805be45: Already exists 
f7869fb3b98c: Already exists 
f75000414ffb: Already exists 
c27c5092058d: Already exists 
add65c7404ea: Already exists 
10f450dd5624: Pull complete 
Digest: sha256:93d336ff26747d0e9db44c86f70fe6f1a0ce059b5bbb756492c3781cb361b35b
Status: Downloaded newer image for casparcg/boost:1.66
 ---> 343b4ebc40a7
Step 3/15 : COPY --from=casparcg/ffmpeg:3.4.1 /opt/ffmpeg /opt/ffmpeg
3.4.1: Pulling from casparcg/ffmpeg
18c8f805be45: Already exists 
f7869fb3b98c: Already exists 
f75000414ffb: Already exists 
c27c5092058d: Already exists 
add65c7404ea: Already exists 
752a2c057cb9: Pull complete 
Digest: sha256:2eb59c521c241074e331ee61c2e70f38b4881298742197be8c07e1a6320f10d8
Status: Downloaded newer image for casparcg/ffmpeg:3.4.1
 ---> 16891c39054b
Step 4/15 : COPY --from=casparcg/cef:3.3239.1723 /opt/cef /opt/cef
3.3239.1723: Pulling from casparcg/cef
18c8f805be45: Already exists 
f7869fb3b98c: Already exists 
f75000414ffb: Already exists 
c27c5092058d: Already exists 
add65c7404ea: Already exists 
33081f3bc0a5: Pull complete 
Digest: sha256:7aea85034ee5b45b6d804cdee9ed62de71902e960d624aa6f731d0b11eeafd1f
Status: Downloaded newer image for casparcg/cef:3.3239.1723
 ---> a59c1bb997f8
Step 5/15 : RUN mkdir /source && mkdir /build && mkdir /install
 ---> Running in aeeb3e6e3ccc
Removing intermediate container aeeb3e6e3ccc
 ---> f9ec2652342c
Step 6/15 : COPY ./src /source
 ---> f062934a03c5
Step 7/15 : WORKDIR /build
Removing intermediate container 4009bb5574a0
 ---> 9b4c73d5bb11
Step 8/15 : ENV BOOST_ROOT=/opt/boost
 ---> Running in 62f537539831
Removing intermediate container 62f537539831
 ---> 4cbca527e33c
Step 9/15 : ENV PKG_CONFIG_PATH=/opt/ffmpeg/lib/pkgconfig
 ---> Running in 9889d6b43998
Removing intermediate container 9889d6b43998
 ---> f6c03d66c4ff
Step 10/15 : RUN cmake /source
 ---> Running in f2127011c422
-- The C compiler identification is GNU 7.2.0
-- The CXX compiler identification is GNU 7.2.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Setting build type to 'Release' as none was specified.
-- Found Git: /usr/bin/git (found version "2.14.1") 
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Boost version: 1.66.0
-- Found the following Boost libraries:
--   system
--   thread
--   chrono
--   filesystem
--   log
--   locale
--   regex
--   date_time
--   coroutine
--   atomic
--   log_setup
--   context
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.29.1") 
-- Checking for module 'libavcodec'
--   Found libavcodec, version 57.107.100
-- Checking for module 'libavformat'
--   Found libavformat, version 57.83.100
-- Checking for module 'libavdevice'
--   Found libavdevice, version 57.10.100
-- Checking for module 'libavutil'
--   Found libavutil, version 55.78.100
-- Checking for module 'libavfilter'
--   Found libavfilter, version 6.107.100
-- Checking for module 'libswscale'
--   Found libswscale, version 4.8.100
-- Checking for module 'libpostproc'
--   Found libpostproc, version 54.7.100
-- Checking for module 'libswresample'
--   Found libswresample, version 2.9.100
-- Found FFmpeg: avcodec;webp;rtmp;z;gmp;ass;ssl;crypto;xcb;xcb-shm;xcb-xfixes;xcb-shape;ssl;crypto;x265;x264;webpmux;webp;vpx;m;vpx;m;vpx;m;vpx;m;vorbisenc;vorbis;theoraenc;theoradec;ogg;rtmp;z;gmp;opus;mp3lame;freetype;fdk-aac;ass;m;dl;lzma;bz2;z;swresample;m;avutil;m;avformat;webp;rtmp;z;gmp;ass;ssl;crypto;xcb;xcb-shm;xcb-xfixes;xcb-shape;ssl;crypto;x265;x264;webpmux;webp;vpx;m;vpx;m;vpx;m;vpx;m;vorbisenc;vorbis;theoraenc;theoradec;ogg;rtmp;z;gmp;opus;mp3lame;freetype;fdk-aac;ass;m;dl;lzma;bz2;z;avcodec;webp;rtmp;z;gmp;ass;ssl;crypto;xcb;xcb-shm;xcb-xfixes;xcb-shape;ssl;crypto;x265;x264;webpmux;webp;vpx;m;vpx;m;vpx;m;vpx;m;vorbisenc;vorbis;theoraenc;theoradec;ogg;rtmp;z;gmp;opus;mp3lame;freetype;fdk-aac;ass;m;dl;lzma;bz2;z;swresample;m;avutil;m;avutil;m;avdevice;webp;rtmp;z;gmp;ass;ssl;crypto;xcb;xcb-shm;xcb-xfixes;xcb-shape;ssl;crypto;x265;x264;webpmux;webp;vpx;m;vpx;m;vpx;m;vpx;m;vorbisenc;vorbis;theoraenc;theoradec;ogg;rtmp;z;gmp;opus;mp3lame;freetype;fdk-aac;ass;m;dl;lzma;bz2;z;avfilter;webp;rtmp;z;gmp;ass;ssl;crypto;xcb;xcb-shm;xcb-xfixes;xcb-shape;ssl;crypto;x265;x264;webpmux;webp;vpx;m;vpx;m;vpx;m;vpx;m;vorbisenc;vorbis;theoraenc;theoradec;ogg;rtmp;z;gmp;opus;mp3lame;freetype;fdk-aac;ass;m;dl;lzma;bz2;z;avresample;m;swscale;m;postproc;avformat;webp;rtmp;z;gmp;ass;ssl;crypto;xcb;xcb-shm;xcb-xfixes;xcb-shape;ssl;crypto;x265;x264;webpmux;webp;vpx;m;vpx;m;vpx;m;vpx;m;vorbisenc;vorbis;theoraenc;theoradec;ogg;rtmp;z;gmp;opus;mp3lame;freetype;fdk-aac;ass;m;dl;lzma;bz2;z;avcodec;webp;rtmp;z;gmp;ass;ssl;crypto;xcb;xcb-shm;xcb-xfixes;xcb-shape;ssl;crypto;x265;x264;webpmux;webp;vpx;m;vpx;m;vpx;m;vpx;m;vorbisenc;vorbis;theoraenc;theoradec;ogg;rtmp;z;gmp;opus;mp3lame;freetype;fdk-aac;ass;m;dl;lzma;bz2;z;swresample;m;avutil;m;avfilter;webp;rtmp;z;gmp;ass;ssl;crypto;xcb;xcb-shm;xcb-xfixes;xcb-shape;ssl;crypto;x265;x264;webpmux;webp;vpx;m;vpx;m;vpx;m;vpx;m;vorbisenc;vorbis;theoraenc;theoradec;ogg;rtmp;z;gmp;opus;mp3lame;freetype;fdk-aac;ass;m;dl;lzma;bz2;z;avresample;m;swscale;m;postproc;avformat;webp;rtmp;z;gmp;ass;ssl;crypto;xcb;xcb-shm;xcb-xfixes;xcb-shape;ssl;crypto;x265;x264;webpmux;webp;vpx;m;vpx;m;vpx;m;vpx;m;vorbisenc;vorbis;theoraenc;theoradec;ogg;rtmp;z;gmp;opus;mp3lame;freetype;fdk-aac;ass;m;dl;lzma;bz2;z;avcodec;webp;rtmp;z;gmp;ass;ssl;crypto;xcb;xcb-shm;xcb-xfixes;xcb-shape;ssl;crypto;x265;x264;webpmux;webp;vpx;m;vpx;m;vpx;m;vpx;m;vorbisenc;vorbis;theoraenc;theoradec;ogg;rtmp;z;gmp;opus;mp3lame;freetype;fdk-aac;ass;m;dl;lzma;bz2;z;swresample;m;avutil;m;postproc;avutil;m;swscale;m;avutil;m;swresample;m;avutil;m  
-- Found OpenGL: /usr/lib/x86_64-linux-gnu/libGL.so  
-- Found JPEG: /usr/lib/x86_64-linux-gnu/libjpeg.so  
-- Looking for FreeImage
--   libraries : /usr/lib/x86_64-linux-gnu/libfreeimage.so from 
--   includes  : /usr/include
-- Found Freetype: /usr/lib/x86_64-linux-gnu/libfreetype.so (found version "2.8.0") 
-- Found GLEW: /usr/include  
-- Found libsndfile: /usr/lib/x86_64-linux-gnu/libsndfile.so
-- Found OpenAL: /usr/lib/x86_64-linux-gnu/libopenal.so  
-- Looking for XOpenDisplay in /usr/lib/x86_64-linux-gnu/libX11.so;/usr/lib/x86_64-linux-gnu/libXext.so
-- Looking for XOpenDisplay in /usr/lib/x86_64-linux-gnu/libX11.so;/usr/lib/x86_64-linux-gnu/libXext.so - found
-- Looking for gethostbyname
-- Looking for gethostbyname - found
-- Looking for connect
-- Looking for connect - found
-- Looking for remove
-- Looking for remove - found
-- Looking for shmat
-- Looking for shmat - found
-- Looking for IceConnectionNumber in ICE
-- Looking for IceConnectionNumber in ICE - found
-- Found X11: /usr/lib/x86_64-linux-gnu/libX11.so
-- Found GLFW: /usr/include (found version "3.2.1") 
-- Found SFML 2.4.2 in /usr/include
-- Unsupported target platform, will still attempt build!
-- Target platform: ubuntu17.10
-- Target architecture: x86_64
-- CPU Count: 12
-- Target build type: Release
-- Build Version: 2.2.0 Beta 1 (Revision: fatal: Not a git repository (or any of the parent directories): .git-fatal: Not a git repository (or any of the parent directories): .git)
ogl/image/image_kernel.cpp"-Winvalid-pch" "-include" "/build/accelerator/accelerator_pch/StdAfx.h"
ogl/image/image_mixer.cpp"-Winvalid-pch" "-include" "/build/accelerator/accelerator_pch/StdAfx.h"
ogl/image/image_shader.cpp"-Winvalid-pch" "-include" "/build/accelerator/accelerator_pch/StdAfx.h"
ogl/util/buffer.cpp"-Winvalid-pch" "-include" "/build/accelerator/accelerator_pch/StdAfx.h"
ogl/util/device.cpp"-Winvalid-pch" "-include" "/build/accelerator/accelerator_pch/StdAfx.h"
ogl/util/shader.cpp"-Winvalid-pch" "-include" "/build/accelerator/accelerator_pch/StdAfx.h"
ogl/util/texture.cpp"-Winvalid-pch" "-include" "/build/accelerator/accelerator_pch/StdAfx.h"
accelerator.cpp"-Winvalid-pch" "-include" "/build/accelerator/accelerator_pch/StdAfx.h"
StdAfx.cpp"-Winvalid-pch" "-include" "/build/accelerator/accelerator_pch/StdAfx.h"
diagnostics/graph.cpp"-Winvalid-pch" "-include" "/build/common/common_pch/stdafx.h"
gl/gl_check.cpp"-Winvalid-pch" "-include" "/build/common/common_pch/stdafx.h"
base64.cpp"-Winvalid-pch" "-include" "/build/common/common_pch/stdafx.h"
env.cpp"-Winvalid-pch" "-include" "/build/common/common_pch/stdafx.h"
filesystem.cpp"-Winvalid-pch" "-include" "/build/common/common_pch/stdafx.h"
log.cpp"-Winvalid-pch" "-include" "/build/common/common_pch/stdafx.h"
stdafx.cpp"-Winvalid-pch" "-include" "/build/common/common_pch/stdafx.h"
tweener.cpp"-Winvalid-pch" "-include" "/build/common/common_pch/stdafx.h"
utf.cpp"-Winvalid-pch" "-include" "/build/common/common_pch/stdafx.h"
os/linux/filesystem.cpp"-Winvalid-pch" "-include" "/build/common/common_pch/stdafx.h"
os/linux/prec_timer.cpp"-Winvalid-pch" "-include" "/build/common/common_pch/stdafx.h"
os/linux/thread.cpp"-Winvalid-pch" "-include" "/build/common/common_pch/stdafx.h"
consumer/frame_consumer.cpp"-Winvalid-pch" "-include" "/build/core/core_pch/StdAfx.h"
consumer/output.cpp"-Winvalid-pch" "-include" "/build/core/core_pch/StdAfx.h"
diagnostics/call_context.cpp"-Winvalid-pch" "-include" "/build/core/core_pch/StdAfx.h"
diagnostics/osd_graph.cpp"-Winvalid-pch" "-include" "/build/core/core_pch/StdAfx.h"
frame/draw_frame.cpp"-Winvalid-pch" "-include" "/build/core/core_pch/StdAfx.h"
frame/frame.cpp"-Winvalid-pch" "-include" "/build/core/core_pch/StdAfx.h"
frame/frame_transform.cpp"-Winvalid-pch" "-include" "/build/core/core_pch/StdAfx.h"
frame/geometry.cpp"-Winvalid-pch" "-include" "/build/core/core_pch/StdAfx.h"
mixer/audio/audio_mixer.cpp"-Winvalid-pch" "-include" "/build/core/core_pch/StdAfx.h"
mixer/image/blend_modes.cpp"-Winvalid-pch" "-include" "/build/core/core_pch/StdAfx.h"
mixer/mixer.cpp"-Winvalid-pch" "-include" "/build/core/core_pch/StdAfx.h"
producer/color/color_producer.cpp"-Winvalid-pch" "-include" "/build/core/core_pch/StdAfx.h"
producer/separated/separated_producer.cpp"-Winvalid-pch" "-include" "/build/core/core_pch/StdAfx.h"
producer/transition/transition_producer.cpp"-Winvalid-pch" "-include" "/build/core/core_pch/StdAfx.h"
producer/route/route_producer.cpp"-Winvalid-pch" "-include" "/build/core/core_pch/StdAfx.h"
producer/cg_proxy.cpp"-Winvalid-pch" "-include" "/build/core/core_pch/StdAfx.h"
producer/frame_producer.cpp"-Winvalid-pch" "-include" "/build/core/core_pch/StdAfx.h"
producer/layer.cpp"-Winvalid-pch" "-include" "/build/core/core_pch/StdAfx.h"
producer/stage.cpp"-Winvalid-pch" "-include" "/build/core/core_pch/StdAfx.h"
StdAfx.cpp"-Winvalid-pch" "-include" "/build/core/core_pch/StdAfx.h"
video_channel.cpp"-Winvalid-pch" "-include" "/build/core/core_pch/StdAfx.h"
video_format.cpp"-Winvalid-pch" "-include" "/build/core/core_pch/StdAfx.h"
producer/av_producer.cpp"-Winvalid-pch" "-include" "/build/modules/ffmpeg/ffmpeg_pch/StdAfx.h"
producer/av_input.cpp"-Winvalid-pch" "-include" "/build/modules/ffmpeg/ffmpeg_pch/StdAfx.h"
util/av_util.cpp"-Winvalid-pch" "-include" "/build/modules/ffmpeg/ffmpeg_pch/StdAfx.h"
producer/ffmpeg_producer.cpp"-Winvalid-pch" "-include" "/build/modules/ffmpeg/ffmpeg_pch/StdAfx.h"
consumer/ffmpeg_consumer.cpp"-Winvalid-pch" "-include" "/build/modules/ffmpeg/ffmpeg_pch/StdAfx.h"
ffmpeg.cpp"-Winvalid-pch" "-include" "/build/modules/ffmpeg/ffmpeg_pch/StdAfx.h"
StdAfx.cpp"-Winvalid-pch" "-include" "/build/modules/ffmpeg/ffmpeg_pch/StdAfx.h"
consumer/decklink_consumer.cpp"-Winvalid-pch" "-include" "/build/modules/decklink/decklink_pch/StdAfx.h"
producer/decklink_producer.cpp"-Winvalid-pch" "-include" "/build/modules/decklink/decklink_pch/StdAfx.h"
decklink.cpp"-Winvalid-pch" "-include" "/build/modules/decklink/decklink_pch/StdAfx.h"
StdAfx.cpp"-Winvalid-pch" "-include" "/build/modules/decklink/decklink_pch/StdAfx.h"
linux_interop/DeckLinkAPIDispatch.cpp"-Winvalid-pch" "-include" "/build/modules/decklink/decklink_pch/StdAfx.h"
amcp/AMCPCommandQueue.cpp"-Winvalid-pch" "-include" "/build/protocol/protocol_pch/StdAfx.h"
amcp/AMCPCommandsImpl.cpp"-Winvalid-pch" "-include" "/build/protocol/protocol_pch/StdAfx.h"
amcp/AMCPProtocolStrategy.cpp"-Winvalid-pch" "-include" "/build/protocol/protocol_pch/StdAfx.h"
amcp/amcp_command_repository.cpp"-Winvalid-pch" "-include" "/build/protocol/protocol_pch/StdAfx.h"
cii/CIICommandsImpl.cpp"-Winvalid-pch" "-include" "/build/protocol/protocol_pch/StdAfx.h"
cii/CIIProtocolStrategy.cpp"-Winvalid-pch" "-include" "/build/protocol/protocol_pch/StdAfx.h"
clk/CLKProtocolStrategy.cpp"-Winvalid-pch" "-include" "/build/protocol/protocol_pch/StdAfx.h"
clk/clk_commands.cpp"-Winvalid-pch" "-include" "/build/protocol/protocol_pch/StdAfx.h"
clk/clk_command_processor.cpp"-Winvalid-pch" "-include" "/build/protocol/protocol_pch/StdAfx.h"
osc/oscpack/OscOutboundPacketStream.cpp"-Winvalid-pch" "-include" "/build/protocol/protocol_pch/StdAfx.h"
osc/oscpack/OscPrintReceivedElements.cpp"-Winvalid-pch" "-include" "/build/protocol/protocol_pch/StdAfx.h"
osc/oscpack/OscReceivedElements.cpp"-Winvalid-pch" "-include" "/build/protocol/protocol_pch/StdAfx.h"
osc/oscpack/OscTypes.cpp"-Winvalid-pch" "-include" "/build/protocol/protocol_pch/StdAfx.h"
osc/client.cpp"-Winvalid-pch" "-include" "/build/protocol/protocol_pch/StdAfx.h"
util/AsyncEventServer.cpp"-Winvalid-pch" "-include" "/build/protocol/protocol_pch/StdAfx.h"
util/lock_container.cpp"-Winvalid-pch" "-include" "/build/protocol/protocol_pch/StdAfx.h"
util/strategy_adapters.cpp"-Winvalid-pch" "-include" "/build/protocol/protocol_pch/StdAfx.h"
util/http_request.cpp"-Winvalid-pch" "-include" "/build/protocol/protocol_pch/StdAfx.h"
StdAfx.cpp"-Winvalid-pch" "-include" "/build/protocol/protocol_pch/StdAfx.h"
-- Configuring done
-- Generating done
-- Build files have been written to: /build
Removing intermediate container f2127011c422
 ---> 7e02166b804c
Step 11/15 : RUN make -j8
 ---> Running in 0b4d76425e24
Scanning dependencies of target casparcg_copy_dependencies
[  1%] Updating stdafx.h
[  1%] Built target casparcg_copy_dependencies
[  2%] Precompiling stdafx.h for common (C++)
/build/common/common_pch/stdafx.h:22:9: warning: #pragma once in main file
 #pragma once
         ^~~~
Scanning dependencies of target common
[  2%] Building CXX object common/CMakeFiles/common.dir/diagnostics/graph.cpp.o
[  3%] Building CXX object common/CMakeFiles/common.dir/gl/gl_check.cpp.o
[  4%] Building CXX object common/CMakeFiles/common.dir/base64.cpp.o
[  5%] Building CXX object common/CMakeFiles/common.dir/filesystem.cpp.o
[  6%] Building CXX object common/CMakeFiles/common.dir/env.cpp.o
[  7%] Building CXX object common/CMakeFiles/common.dir/log.cpp.o
[  8%] Building CXX object common/CMakeFiles/common.dir/stdafx.cpp.o
[  9%] Building CXX object common/CMakeFiles/common.dir/tweener.cpp.o
[ 10%] Building CXX object common/CMakeFiles/common.dir/utf.cpp.o
[ 11%] Building CXX object common/CMakeFiles/common.dir/os/linux/filesystem.cpp.o
[ 11%] Building CXX object common/CMakeFiles/common.dir/os/linux/prec_timer.cpp.o
[ 12%] Building CXX object common/CMakeFiles/common.dir/os/linux/thread.cpp.o
[ 13%] Linking CXX static library libcommon.a
[ 13%] Built target common
[ 14%] Updating StdAfx.h
[ 15%] Precompiling StdAfx.h for core (C++)
/build/core/core_pch/StdAfx.h:22:9: warning: #pragma once in main file
 #pragma once
         ^~~~
Scanning dependencies of target core
[ 16%] Building CXX object core/CMakeFiles/core.dir/diagnostics/osd_graph.cpp.o
[ 17%] Building CXX object core/CMakeFiles/core.dir/diagnostics/call_context.cpp.o
[ 18%] Building CXX object core/CMakeFiles/core.dir/consumer/frame_consumer.cpp.o
[ 19%] Building CXX object core/CMakeFiles/core.dir/consumer/output.cpp.o
[ 20%] Building CXX object core/CMakeFiles/core.dir/frame/frame.cpp.o
[ 21%] Building CXX object core/CMakeFiles/core.dir/frame/frame_transform.cpp.o
[ 21%] Building CXX object core/CMakeFiles/core.dir/frame/draw_frame.cpp.o
[ 22%] Building CXX object core/CMakeFiles/core.dir/frame/geometry.cpp.o
[ 23%] Building CXX object core/CMakeFiles/core.dir/mixer/audio/audio_mixer.cpp.o
[ 24%] Building CXX object core/CMakeFiles/core.dir/mixer/image/blend_modes.cpp.o
[ 25%] Building CXX object core/CMakeFiles/core.dir/mixer/mixer.cpp.o
[ 26%] Building CXX object core/CMakeFiles/core.dir/producer/color/color_producer.cpp.o
[ 27%] Building CXX object core/CMakeFiles/core.dir/producer/separated/separated_producer.cpp.o
[ 27%] Building CXX object core/CMakeFiles/core.dir/producer/transition/transition_producer.cpp.o
[ 28%] Building CXX object core/CMakeFiles/core.dir/producer/route/route_producer.cpp.o
[ 29%] Building CXX object core/CMakeFiles/core.dir/producer/cg_proxy.cpp.o
[ 30%] Building CXX object core/CMakeFiles/core.dir/producer/frame_producer.cpp.o
In file included from /source/core/producer/route/route_producer.cpp:30:0:
/source/core/../core/video_channel.h:80:48: error: declaration of 'std::shared_ptr<caspar::core::route> caspar::core::video_channel::route(int)' [-fpermissive]
     std::shared_ptr<route> route(int index = -1);
                                                ^
/source/core/../core/video_channel.h:38:8: error: changes meaning of 'route' from 'struct caspar::core::route' [-fpermissive]
 struct route
        ^~~~~
In file included from /source/core/producer/cg_proxy.cpp:29:0:
/source/core/producer/../video_channel.h:80:48: error: declaration of 'std::shared_ptr<caspar::core::route> caspar::core::video_channel::route(int)' [-fpermissive]
     std::shared_ptr<route> route(int index = -1);
                                                ^
/source/core/producer/../video_channel.h:38:8: error: changes meaning of 'route' from 'struct caspar::core::route' [-fpermissive]
 struct route
        ^~~~~
[ 31%] Building CXX object core/CMakeFiles/core.dir/producer/layer.cpp.o
[ 32%] Building CXX object core/CMakeFiles/core.dir/producer/stage.cpp.o
[ 33%] Building CXX object core/CMakeFiles/core.dir/StdAfx.cpp.o
[ 34%] Building CXX object core/CMakeFiles/core.dir/video_channel.cpp.o
make[2]: *** [core/CMakeFiles/core.dir/producer/cg_proxy.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
core/CMakeFiles/core.dir/build.make:463: recipe for target 'core/CMakeFiles/core.dir/producer/cg_proxy.cpp.o' failed
make[2]: *** [core/CMakeFiles/core.dir/producer/route/route_producer.cpp.o] Error 1
core/CMakeFiles/core.dir/build.make:437: recipe for target 'core/CMakeFiles/core.dir/producer/route/route_producer.cpp.o' failed
In file included from /source/core/video_channel.cpp:24:0:
/source/core/video_channel.h:80:48: error: declaration of 'std::shared_ptr<caspar::core::route> caspar::core::video_channel::route(int)' [-fpermissive]
     std::shared_ptr<route> route(int index = -1);
                                                ^
/source/core/video_channel.h:38:8: error: changes meaning of 'route' from 'struct caspar::core::route' [-fpermissive]
 struct route
        ^~~~~
make[2]: *** [core/CMakeFiles/core.dir/video_channel.cpp.o] Error 1
core/CMakeFiles/core.dir/build.make:593: recipe for target 'core/CMakeFiles/core.dir/video_channel.cpp.o' failed
make[1]: *** [core/CMakeFiles/core.dir/all] Error 2
CMakeFiles/Makefile2:197: recipe for target 'core/CMakeFiles/core.dir/all' failed
make: *** [all] Error 2
Makefile:83: recipe for target 'all' failed
The command '/bin/sh -c make -j8' returned a non-zero code: 2

I've fixed it. CEF still has issues though as @Julusian pointed out.

```
[ 47%] Building CXX object modules/html/CMakeFiles/html.dir/html.cpp.o
/source/modules/html/producer/html_producer.cpp:54:10: fatal error: include/cef_app.h: No such file or directory
 #include <include/cef_app.h>
          ^~~~~~~~~~~~~~~~~~~
compilation terminated.
modules/html/CMakeFiles/html.dir/build.make:86: recipe for target 'modules/html/CMakeFiles/html.dir/producer/html_producer.cpp.o' failed
make[2]: *** [modules/html/CMakeFiles/html.dir/producer/html_producer.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
Scanning dependencies of target protocol
/source/modules/html/html.cpp:44:10: fatal error: include/cef_app.h: No such file or directory
 #include <include/cef_app.h>
          ^~~~~~~~~~~~~~~~~~~
```

```
/source/modules/decklink/consumer/decklink_consumer.cpp:284:54: error: BMDFieldDominance is not a class, namespace, or enumeration
     int field_count_ = mode_->GetFieldDominance() != BMDFieldDominance::bmdProgressiveFrame ? 2 : 1;
                                                      ^~~~~~~~~~~~~~~~~
/source/modules/decklink/consumer/decklink_consumer.cpp: In member function virtual HRESULT caspar::decklink::decklink_consumer::ScheduledFrameCompleted(IDeckLinkVideoFrame*, BMDOutputFrameCompletionResult):
/source/modules/decklink/consumer/decklink_consumer.cpp:473:47: error: 'BMDFieldDominance is not a class, namespace, or enumeration
             if (mode_->GetFieldDominance() != BMDFieldDominance::bmdProgressiveFrame) {
                                               ^~~~~~~~~~~~~~~~~
/source/modules/decklink/consumer/decklink_consumer.cpp:474:51: error: 'BMDFieldDominance is not a class, namespace, or enumeration
                 if (mode_->GetFieldDominance() != BMDFieldDominance::bmdUpperFieldFirst) {
                                                   ^~~~~~~~~~~~~~~~~
/source/modules/decklink/producer/decklink_producer.cpp:283:54: error: 'BMDFieldDominance is not a class, namespace, or enumeration
     int field_count_ = mode_->GetFieldDominance() != BMDFieldDominance::bmdProgressiveFrame ? 2 : 1;
                                                      ^~~~~~~~~~~~~~~~~
/source/modules/decklink/producer/decklink_producer.cpp: In member function virtual HRESULT caspar::decklink::decklink_producer::VideoInputFrameArrived(IDeckLinkVideoInputFrame*, IDeckLinkAudioInputPacket*):
/source/modules/decklink/producer/decklink_producer.cpp:387:71: error: BMDFieldDominance is not a class, namespace, or enumeration
                 src->interlaced_frame = mode_->GetFieldDominance() != BMDFieldDominance::bmdProgressiveFrame;
                                                                       ^~~~~~~~~~~~~~~~~
/source/modules/decklink/producer/decklink_producer.cpp:388:71: error: BMDFieldDominance is not a class, namespace, or enumeration
                 src->top_field_first  = mode_->GetFieldDominance() == BMDFieldDominance::bmdUpperFieldFirst ? 1 : 0;

```
I'm guessing one of the commits of the past 2-3 days borked something...

--- a/src/modules/decklink/consumer/decklink_consumer.cpp
+++ b/src/modules/decklink/consumer/decklink_consumer.cpp
@@ -281,7 +281,7 @@ struct decklink_consumer : public IDeckLinkVideoOutputCallback
 
     com_ptr<IDeckLinkDisplayMode> mode_ =
         get_display_mode(output_, format_desc_.format, bmdFormat8BitBGRA, bmdVideoOutputFlagDefault);
-    int field_count_ = mode_->GetFieldDominance() != BMDFieldDominance::bmdProgressiveFrame ? 2 : 1;
+    int field_count_ = mode_->GetFieldDominance() != bmdProgressiveFrame ? 2 : 1;
 
     std::atomic<bool> abort_request_{false};
 
@@ -470,8 +470,8 @@ struct decklink_consumer : public IDeckLinkVideoOutputCallback
                 buffer_cond_.notify_all();
             }
 
-            if (mode_->GetFieldDominance() != BMDFieldDominance::bmdProgressiveFrame) {
-                if (mode_->GetFieldDominance() != BMDFieldDominance::bmdUpperFieldFirst) {
+            if (mode_->GetFieldDominance() != bmdProgressiveFrame) {
+                if (mode_->GetFieldDominance() != bmdUpperFieldFirst) {
                     std::swap(frames[0], frames[1]);
                 }
 

--- a/src/modules/decklink/producer/decklink_producer.cpp
+++ b/src/modules/decklink/producer/decklink_producer.cpp
@@ -280,7 +280,7 @@ class decklink_producer : public IDeckLinkInputCallback
 
     com_ptr<IDeckLinkDisplayMode> mode_ =
         get_display_mode(input_, format_desc_.format, bmdFormat8BitBGRA, bmdVideoOutputFlagDefault);
-    int field_count_ = mode_->GetFieldDominance() != BMDFieldDominance::bmdProgressiveFrame ? 2 : 1;
+    int field_count_ = mode_->GetFieldDominance() != bmdProgressiveFrame ? 2 : 1;
 
     Filter video_filter_;
     Filter audio_filter_;
@@ -384,8 +384,8 @@ class decklink_producer : public IDeckLinkInputCallback
                 src->format = AV_PIX_FMT_UYVY422;
                 src->width  = video->GetWidth();
                 src->height = video->GetHeight();
-                src->interlaced_frame = mode_->GetFieldDominance() != BMDFieldDominance::bmdProgressiveFrame;
-                src->top_field_first  = mode_->GetFieldDominance() == BMDFieldDominance::bmdUpperFieldFirst ? 1 : 0;
+                src->interlaced_frame = mode_->GetFieldDominance() != bmdProgressiveFrame;
+                src->top_field_first  = mode_->GetFieldDominance() == bmdUpperFieldFirst ? 1 : 0;
                 src->key_frame        = 1;
 
                 void* video_bytes = nullptr;

--- a/src/modules/html/CMakeLists.txt
+++ b/src/modules/html/CMakeLists.txt
@@ -22,6 +22,7 @@ include_directories(../..)
 include_directories(${BOOST_INCLUDE_PATH})
 include_directories(${TBB_INCLUDE_PATH})
 include_directories(${CEF_INCLUDE_PATH})
+include_directories(${CEF_PATH})
 
 set_target_properties(html PROPERTIES FOLDER modules)
 source_group(sources\\producer producer/*)
@@ -45,6 +46,17 @@ else()
 		${CEF_BIN_PATH}/libcef.so
 		libcef_dll_wrapper
 	)
+
+	# Build and link in libcef_dll_wrapper.
+	set(CEF_ROOT "${CEF_PATH}")
+	set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CEF_PATH}/cmake")
+	find_package(CEF REQUIRED)
+	add_subdirectory(${CEF_LIBCEF_DLL_WRAPPER_PATH} libcef_dll_wrapper)
+
+	# libcef_dll_wrapper exports a set of compiler flags that are not
+	# compatible with e.g. Boost.Log, so we need to reset them here.
+	SET_TARGET_PROPERTIES(libcef_dll_wrapper PROPERTIES INTERFACE_COMPILE_OPTIONS "")
+
 endif()
 
 casparcg_add_include_statement("modules/html/html.h")

